openpype/modules/kitsu/utils/update_op_with_zou.py

```diff
@@ -162,14 +179,32 @@ def update_op_assets(
         )
         if visual_parent_doc_id is None:
             # Find root folder doc
-            root_folder_doc = dbcon.find_one(
-                {
-                    "type": "asset",
-                    "name": entity_parent_folders[-1],
-                    "data.root_of": substitute_item_type,
-                },
-                ["_id"],
+            root_folder_docs = get_assets(
+                project_name,
+                asset_name=[entity_parent_folders[-1]],
+                fields=["_id", "data.root_of"]
             )
+            # NOTE: Not sure why it's checking for entity type?
```

Comment:
If I remember correctly, it is for sorting by `Assets` or `Shots`

Reply:
But if you would have `Bob` in `Shots` and `Assets` OpenPype would not be able to work properly because it would be duplicated asset name.

Long store short: If there is and asset with same name in OpenPype there are 2 possible:
1. There is duplicated name in project which can't happen and should skip it's creation
2. It is the same entity but was moved from `Shots` to `Assets` (for example) so should update the type
In both cases it should not create new asset document with same name.

Reply:
>  I don't understand why you talk about same name issue here.

Because this code here can cause the duplicated name issue.

The logic here is:
Query asset with name <NAME> and type <TYPE> if was not found then create the asset.

But, what if the asset with the <NAME> already exists but does not have the <TYPE>? In that case you will create second asset document with name <NAME> and from that moment OpenPype won't work correctly.

Reply:
Similar issue to this https://github.com/pypeclub/OpenPype/pull/3496#discussion_r919721371

Maybe just skip the check of `root_of`, query the asset byt name and it would do the same thing.